backends/tfhe-cuda-backend/cuda/include/integer/vector_find.h

```diff
@@ -0,0 +1,362 @@
+#pragma once
+#include "integer/comparison.h"
+#include "integer/radix_ciphertext.cuh"
+#include "integer_utilities.h"
+#include <functional>
+#include <vector>
+
+template <typename Torus> struct int_equality_selectors_buffer {
+  int_radix_params params;
+  bool allocate_gpu_memory;
+  uint32_t lut_stride;
+
+  uint32_t num_possible_values;
+  int_radix_lut<Torus> *comparison_luts;
+  CudaRadixCiphertextFFI *tmp_many_luts_output;
+
+  CudaRadixCiphertextFFI *tmp_block_comparisons;
+  int_comparison_buffer<Torus> *reduction_buffer;
+
+  int_equality_selectors_buffer(CudaStreams streams, int_radix_params params,
+                                uint32_t num_possible_values,
+                                uint32_t num_blocks, bool allocate_gpu_memory,
+                                uint64_t &size_tracker) {
+    this->params = params;
+    this->allocate_gpu_memory = allocate_gpu_memory;
+    this->num_possible_values = num_possible_values;
+
+    uint32_t N = num_blocks;
```

Comment:
N is usually used to denote the polynomial size. Also, we use lower case variables everywhere (except for Class names or Enums). Here I don't think there is a need to have a variable to wrap num_blocks you could probably just remove it.